addons/base_import/models/base_import.py

```diff
@@ -1402,6 +1402,8 @@ def execute_import(self, fields, columns, options, dryrun=False):
                 self.pool.clear_all_caches()
                 # don't propagate to other workers since it was rollbacked
                 self.pool.reset_changes()
+                # if anything is marked dirty, changes will actually get written later
+                self.env.cache._dirty.clear()
```

Comment:
I think you should clear everything, not just dirty records.
```suggestion
                self.env.clear()
```